pkg/controller/stack/stack_controller.go

```diff
@@ -1085,6 +1086,85 @@ func (sess *reconcileStackSession) SetEnvRefsForWorkspace(ctx context.Context, w
 	return nil
 }
 
+func (sess *reconcileStackSession) resolveConfigRefs(ctx context.Context) ([]ConfigKeyValue, error) {
+	allConfigs := make([]ConfigKeyValue, 0)
+	for _, configRefMap := range sess.stack.ConfigRefs {
+		for k, ref := range configRefMap {
```

Comment:
I'm not clear on why this is a `[]map[string]ConfigRef` instead of just a `[]ConfigRef`. The key only seems to be used for secret refs (which mirrors `SecretRefs`?), and for the default case (do we need that?).

Reply:
It was a design decision, that I've mentioned here: https://github.com/pulumi/pulumi-kubernetes-operator/pull/575#discussion_r1603536059

Reply:
I've chosen the `array of maps` option to be similar to the `envRefs` and `secretRefs` fields, where the key is the env/secret key name. 

But in fact, for `configRefs`, the key is not used in the code for `literal` and `configMap` options. But I don't know what could be a better option 😕

Reply:
For literal values it's used as the config-key name:

```yaml
configRefs:
   - sample:
        type: Literal
        value: just-a-value
```

it will be resolved as a config like that:

```
sample: just-a-value
```